server/src/main/java/org/opensearch/search/backpressure/trackers/TaskResourceUsageTrackers.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.backpressure.trackers;
+
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.tasks.CancellableTask;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskCancellation;
+
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+/**
+ * TaskResourceUsageTrackers is used to hold all the {@link TaskResourceUsageTracker} objects.
+ *
+ * @opensearch.internal
+ */
+public class TaskResourceUsageTrackers {
+    private TaskResourceUsageTracker cpuUsageTracker;
+    private TaskResourceUsageTracker heapUsageTracker;
+    private TaskResourceUsageTracker elapsedTimeTracker;
+    private final EnumMap<TaskResourceUsageTrackerType, TaskResourceUsageTracker> all;
+
+    public TaskResourceUsageTrackers() {
+        all = new EnumMap<>(TaskResourceUsageTrackerType.class);
+    }
+
+    /**
+     * adds the cpuUsageTracker
+     * @param cpuUsageTracker
+     */
+    public void addCpuUsageTracker(final TaskResourceUsageTracker cpuUsageTracker) {
+        this.cpuUsageTracker = cpuUsageTracker;
+        all.put(TaskResourceUsageTrackerType.CPU_USAGE_TRACKER, cpuUsageTracker);
+    }
+
+    /**
+     * adds the heapUsageTracker
+     * @param heapUsageTracker
+     */
+    public void addHeapUsageTracker(final TaskResourceUsageTracker heapUsageTracker) {
+        this.heapUsageTracker = heapUsageTracker;
+        all.put(TaskResourceUsageTrackerType.HEAP_USAGE_TRACKER, heapUsageTracker);
+    }
+
+    /**
+     * adds the elapsedTimeTracker
+     * @param elapsedTimeTracker
+     */
+    public void addElapsedTimeTracker(final TaskResourceUsageTracker elapsedTimeTracker) {
+        this.elapsedTimeTracker = elapsedTimeTracker;
+        all.put(TaskResourceUsageTrackerType.ELAPSED_TIME_TRACKER, elapsedTimeTracker);
+    }
```

Comment:
similar refactoring (as NodeDuressTrackers) should be done for this class to make it resource agnostic. Resource should be used as parameter instead of type

Reply:
Makes sense